src/Serializer/ItemNormalizer.php

```diff
@@ -37,11 +39,17 @@ public function normalize($object, $format = null, array $context = [])
 
     /**
      * {@inheritdoc}
+     *
+     * @throws InvalidArgumentException
      */
     public function denormalize($data, $class, $format = null, array $context = [])
     {
         // Avoid issues with proxies if we populated the object
         if (isset($data['id']) && !isset($context['object_to_populate'])) {
+            if (isset($context['allow_update']) && true !== $context['allow_update']) {
```

Comment:
I'm uncomfortable with unprefixed `allow_update` because this is the Symfony Serializer context which can contain anything (same for the existing keys we use, yes). What about `api_allow_update`?